### ISSUE TITLE
Non ascii operators are rejected by `identifier_name` for not starting with a lowercase letter.

### ISSUE THREAD

**Issue by ryanbooker** [realm/SwiftLint#1762]
While `["/", "=", "-", "+", "!", "*", "|", "^", "~", "?", ".", "%", "<", ">", "&"]` are specifically allowed as the first character of operators, anything else is rejected.

e.g. using `•` as the compose operator or `§` as function application is relatively common and useful. Including them in `allowed_symbols` doesn't circumvent this behaviour. Should it? The `isOperator` function seems otherwise very narrow in its permissiveness.

I don't really want to disable the entire feature just to allow these functions.

**Comment by marcelofabri** [comment]
> Including them in `allowed_symbols` doesn't circumvent this behaviour. Should it? 

I think so.

---

Also, I think there's another bug in the implementation currently:

```swift
let containsAllowedSymbol = configuration.allowedSymbols.contains(where: name.contains)
                if !containsAllowedSymbol &&
                    !CharacterSet.alphanumerics.isSuperset(ofCharactersIn: name) 
```

This doesn't seem to guarantee that all non-alphanumeric characters in `name` are in `configuration.allowedSymbols`.

**Comment by ryanbooker** [comment]
OK. Cheers. I'll work on something for this in the next couple of days.

**Comment by dcutting** [comment]
Was there any progress on this? I'm trying to permit variables/functions that begin with underscores.

**Comment by SDGGiesbrecht** [comment]
@dcutting, [this comment](https://github.com/realm/SwiftLint/pull/1865#issuecomment-342999424) in the related pull request describes the current state of things.

**Comment by mylemans** [comment]
We are running in the same issue.

**Comment by ryanbooker** [comment]
I don't think I ended up pushing anything. I got sidetracked onto work. :\

**Comment by Sajjon** [comment]
We are running in the same issue.

**Comment by endogenesis** [comment]
allowed_symbols not works for enum cases

**Comment by aklowther** [comment]
has anyone been able to give this attention?

**Comment by colbad2** [comment]
Still an issue

**Comment by billburgess** [comment]
This seems like a huge issue that hasn't been fixed in over 5 years. What the heck?! It seems like allowing characters in your variable, types, etc is a huge oversight. This rule is unusable which means you can't even enforce simple length checks on variables.

**Comment by SimplyDanny** [comment]
The original issue has been fixed by supporting a new option `additional_operators` in the rule's configuration. Just add the symbols you want to have considered as operators to the list and the rule will keep silence on them.

Furthermore, the rule has completely been rewritten so that a few false positives or configuration issues should have been fixed as well. For  anything more, that doesn't work as you expect, please create separate issues.
